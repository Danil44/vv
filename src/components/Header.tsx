import React from 'react';
import { jsx, css } from '@emotion/core';
/**@jsx jsx */

type HeaderProps = {
  shift: boolean;
};

export default function Header({ shift }: HeaderProps) {
  return (
    <header
      css={css`
        padding: 5.188em 5.625em;
        transform: translateX(${shift ? '-70%' : 0});
      `}
    >
      <h2
        css={css`
          color: #fff;
          border-bottom: 1px solid #fff;
          display: inline;
          font-size: 1.25em;
        `}
      >
        FILMER
      </h2>
    </header>
  );
}
