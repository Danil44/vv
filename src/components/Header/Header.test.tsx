import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import { render } from '@testing-library/react';
import Header from './Header';
import { matchers } from 'jest-emotion';

expect.extend(matchers);

describe('<Header />', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<Header shift={false} />);
  });

  it('has not changed', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("shifts left if 'shift' prop is set to true", () => {
    const { getByTestId } = render(<Header shift={false} />);
    expect(getByTestId('header')).toHaveStyleRule('transform', 'translateX(0)');

    wrapper.setProps({ shift: true });

    expect(wrapper.find('header')).toHaveStyleRule(
      'transform',
      'translateX(-70%)',
    );
  });
});
