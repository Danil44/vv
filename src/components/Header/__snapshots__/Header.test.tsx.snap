// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> has not changed 1`] = `ReactWrapper {}`;
